cl/custom_filters/templatetags/extras.py

```diff
@@ -148,3 +151,63 @@ def sort_caret(request, value) -> SafeString:
         if current.startswith("-"):
             caret = '&nbsp;<i class="gray fa fa-angle-down"></i>'
     return mark_safe(caret)
+
+
+@register.simple_tag
+def citation(obj, fmt=None) -> SafeString:
+    if isinstance(obj, Docket):
+        docket = obj
+        date_of_interest = (
+            docket.date_terminated
+            if docket.date_terminated
+            else docket.date_last_filing
+        )
+        ecf = ""
+    elif isinstance(obj, DocketEntry):
+        docket = obj.docket
+        date_of_interest = obj.date_filed
+        ecf = obj.entry_number
+    else:
+        return mark_safe("Unknown object type")
+    if not fmt:
+        fmt = (
+            "{name_bb}, {case_bb} {ecf_bb}, {reporter} ({court_bb} {date_bb})"
+        )
+    # for now we don't know what to use as reporter
+    reporter = ""
+    if date_of_interest:
+        date_of_interest = date_of_interest.strftime("%b %d, %Y")
+    if not docket.court.citation_string:
+        # remove all references to court
+        fmt = re.sub(r"\{court.*?\}\s*", "", fmt)
+    if not date_of_interest:
+        # remove all references to date
+        fmt = re.sub(r"\{date.*?\}\s*", "", fmt)
+    if not reporter:
+        # remove all references to reporter
+        fmt = re.sub(r"\{reporter.*?\}\s*", "", fmt)
+    if not ecf:
+        # remove all references to ecf
+        fmt = re.sub(r"\{ecf.*?\}\s*", "", fmt)
+    fmt = re.sub(r"\(\s*?\)", "", fmt)  # remove empty ()
+    fmt = re.sub(
+        r"\s\s+", " ", fmt
+    )  # change multiple spaces to a single space
+    fmt = re.sub(r",\s?,+", ",", fmt)  # change double comma to a single comma
+    fmt = re.sub(r"\s+$", "", fmt)  # remove trailing whitespace
+    fmt = re.sub(r",+$", "", fmt)  # remove trailing commas
```

Comment:
What are all these cleanup things for? Do we know we need them?

Reply:
Yes, they are needed.  The methodology is to create a format screen with all the potential parts that could be used.  Then we evaluate if those parts are actually going to be used.  If they are not used, then we want to remove those parts along with standard connecting characters.  This can leave excess spaces, commas and other things.

Reply:
OK, so sounds like we can remove/reduce this if we simplify as discussed above, right?